tools/oeedger8r/src/Common.ml

```diff
@@ -42,7 +42,14 @@ let get_parameter_str (p : pdecl) =
 (** ----- End code borrowed and tweaked from {!CodeGen.ml} ----- *)
 
 (* Helper to map and filter out None at the same time. *)
-let filter_map f l = List.of_seq (Seq.filter_map f (List.to_seq l))
+let filter_map f l =
+  let rec apply = function
+    | hd::tl -> (match f hd with
+                | Some v -> v ::(apply tl)
+                | None -> apply tl)
+    | [] -> []
+ in apply l
+
```

Comment:
I'm sorry, but this code looks reused without attribution from [here](https://github.com/ocaml/ocaml/blob/e5ebec705865dd11d86c1dcd1cedb9ace2bcef5d/stdlib/list.ml#L254) (the official OCaml library):

```ocaml
let filter_map f =
  let rec aux accu = function
    | [] -> rev accu
    | x :: l ->
        match f x with
        | None -> aux accu l
        | Some v -> aux (v :: accu) l
  in
  aux []
```

The `accu` arg was dropped, `aux` was renamed to `apply`, and `x` was renamed to `f`, but they are otherwise identical save whitespace. The giveaway to me is the random use of the argument named `v` which is not otherwise used in this file.

The original code is licensed LGPL, and would not be suitable for use here even with attribution.

Reply:
Hmm, is there any other way to implement filter_map? Say if one were to write a `fib` function, more often than not, the code would look identical no?

Reply:
I have avoided uses of `List.filter_map`.